### ISSUE TITLE
Stuck on "software decoding" with Intel graphics

### ISSUE THREAD

**Issue by saintger** [mpv-player/mpv#885]
Hello,

I'm trying to use HW accelerated video with mpv with my Intel GMA HD graphic (device ID 8086-0046) on Debian Testing (amd64).
I made a fresh checkout of mpv-build, enabled vaapi in ffmpeg_options and built mpv without any errors.

However mpv always fallback to "software decoding". I have read:
https://github.com/mpv-player/mpv/issues/372
But it doesn't help.

Here is the output when I use --vo=vaapi --hwdec=vaapi:

```
Playing: test.avi
File tags:
 encoder: MEncoder git-ab94fc6-4.4.3
 title: test.avi
[stream] Video (+) --vid=1 (mpeg4)
[stream] Audio (+) --aid=1 (mp3)
libva info: VA-API version 0.35.1
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_35
libva info: va_openDriver() returns 0
Using software decoding.
AO: [pulse] 48000Hz stereo 2ch s16
AV: 00:00:00 / 00:21:14 (0%) A-V:  0.000
VO: [vaapi] 624x352 => 624x352 yuv420p
```

And here is the output with --vo=opengl --hwdec=vaapi:

```
Playing: test.avi
File tags:
 encoder: MEncoder git-ab94fc6-4.4.3
 title: test.avi
[stream] Video (+) --vid=1 (mpeg4)
[stream] Audio (+) --aid=1 (mp3)
[vo/opengl/x11] X11 error: GLXBadFBConfig
[vo/opengl] Could not create GL3 context. Retrying with legacy context.
Using software decoding.
AO: [pulse] 48000Hz stereo 2ch s16
AV: 00:00:00 / 00:21:14 (0%) A-V:  0.000
VO: [opengl] 624x352 => 624x352 yuv420p
```

As a result, during playback, the CPU load is  quite low (< 10%) but audio and video are not sync.
Do you think that the Intel graphic just does not work or is a sotfware problem ?
Let me know if I can debug this further.

Thanks !

**Comment by n42k** [comment]
For anyone finding this issue later and trying to get hardware acceleration to run on Intel GPUs:

The options above did enable hardware acceleration, but getting rid of the copy part of vaapi-copy made it even better. I went from software decoding (dropping thousands of frames per video), to vaapi-copy hardware decoding (dropping hundreds of frames), to vaapi (dropping a dozen or so frames). Simply use the options `--hwdec=vaapi --hwdec-codecs=all --vo=vaapi`.

I'm using the Intel HD Graphics GPU on the N2840 CPU, and I had to install the following 2 packages on Lubuntu 16.10 32 bits: libvdpau-va-gl1, i965-va-driver.
